src/Toolkit/Html.php

```diff
@@ -446,128 +447,110 @@ public static function video(string $url, array $options = [], array $attr = [])
-            if (preg_match('!' . $schema['pattern'] . '!i', $url, $array) === 1) {
-                $domain = $schema['domain'] ?? $domain;
-                $uri    = $schema['uri'] ?? $uri;
-                $id     = $array[1];
-                if (isset($array[2]) === true) {
-                    parse_str($array[2], $urlOptions);
-
-                    // convert video URL options to embed URL options
-                    if (isset($urlOptions['t']) === true) {
-                        $urlOptions['start'] = $urlOptions['t'];
-                        unset($urlOptions['t']);
-                    }
+        if (preg_match('!youtu!i', $url) !== 1) {
+            throw new Exception('Invalid YouTube source');
+        }
+
+        $uri    = new Uri($url);
+        $path   = $uri->path();
+        $query  = $uri->query();
+        $first  = $path->first();
+        $second = $path->nth(1);
+        $host   = 'https://' . $uri->host() . '/embed';
+        $src    = null;
+
+        $isYoutubeId = function (string $id): bool {
+            return preg_match('!^[a-zA-Z0-9_-]+$!', $id);
+        };
+
+        switch ($path->toString()) {
+            // playlists
+            case 'embed/videoseries':
+            case 'playlist':
+                if ($isYoutubeId($query->list) === true) {
+                    $src = $host . '/videoseries';
+                }
+
+                break;
+
+            // regular video URLs
+            case 'watch':
+                if ($isYoutubeId($query->v) === true) {
+                    $src = $host . '/' . $query->v;
+
+                    $query->start = $query->t;
+
+                    unset($query->v, $query->t);
                 }
+
                 break;
-            }
+
+            default:
+                // short URLs
+                if (Str::contains($uri->host(), 'youtu.be') === true) {
+                    $src = 'https://www.youtube.com/embed/' . $first;
+
+                    $query->start = $query->t;
+                    unset($query->t);
+
+                // embedded video URLs
+                } elseif ($first === 'embed' && $isYoutubeId($second) === true) {
+                    $src = $host . '/' . $second;
+                }
         }
 
-        // no match
-        if ($id === null) {
+        if (empty($src) === true) {
             throw new Exception('Invalid YouTube source');
         }
 
-        // build the options query
-        if (empty($options) === false || empty($urlOptions) === false) {
-            $query = (Str::contains($uri, '?') === true ? '&' : '?') . http_build_query(array_merge($urlOptions, $options));
-        } else {
-            $query = '';
+        // append all query parameters
+        foreach ($options as $key => $value) {
+            $query->$key = $value;
         }
 
-        $url = 'https://' . $domain . '/' . $uri . $id . $query;
+        // allow fullscreen mode by default
+        $attrs = array_merge(['allowfullscreen' => true], $attrs);
```

Comment:
This change (extracting the `array_merge()` into its own line) is not mirrored in the `vimeo()` method. In contrast, the `vimeo()` method already appends the query string. I think the implementation should be mostly consistent for better readability.